packages/dd-trace/src/platform/node/request.js

```diff
@@ -55,8 +132,27 @@ function request (options, callback) {
   req.end()
 }
 
+function getPool (options) {
+  const url = `${options.protocol}//${options.hostname}${options.port ? `:${options.port}` : ''}`
+  let pool = pools[url]
+  if (!pool) {
+    pool = new undici.Pool(url)
+  }
+  return pool
+}
+
 function byteLength (data) {
-  return data.length > 0 ? data.reduce((prev, next) => prev + next.length, 0) : 0
+  let len = 0
+  for (const item of data) {
+    len += item.length
+  }
+  return len
+}
+
+if (undici) {
```

Comment:
Can the 2 implementations be split in multiple files to make the separation clearer?

Reply:
Sure.